sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1621,9 +1621,11 @@ class Analyzer(
 
     /** Extracts a [[Generator]] expression and any names assigned by aliases to their output. */
     private object AliasedGenerator {
-      def unapply(e: Expression): Option[(Generator, Seq[String])] = e match {
-        case Alias(g: Generator, name) if g.resolved => Some((g, name :: Nil))
-        case MultiAlias(g: Generator, names) if g.resolved => Some(g, names)
+      def unapply(e: Expression): Option[(Generator, Seq[String], Boolean)] = e match {
```

Comment:
document what the return value means (especially that boolean value, but also the Seq[String] that's preexisting)